src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2622,3 +2622,15 @@
    # give up, i.e. use the user preference
    .rs.scalar("")
 })
+
+.rs.addJsonRpcHandler("cpp_is_cpp11_file", function(file)
+{
+    if (!file.exists(file))
+        return(.rs.scalar(FALSE))
+
+    lines <- readLines(file)
```

Comment:
nit: use `warn = FALSE` to suppress errors, and consider `tryCatch()` in case the file is unreadable for some other reason (e.g. on Windows, encoding errors might cause translation to fail)